app/code/Magento/SalesGraphQl/Model/CreditmemoItemTypeResolver.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\SalesGraphQl\Model;
+
+use Magento\Framework\GraphQl\Query\Resolver\TypeResolverInterface;
+
+/**
+ * Resolves item type for credit memo item
+ */
+class CreditmemoItemTypeResolver implements TypeResolverInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function resolveType(array $data): string
+    {
+        if (isset($data['product_type'])) {
```

Comment:
I would suggest moving items for different product types to the configuration, for instance, DI.xml, rather hardcoding them.

Reply:
@joni-jones  we actually did this because Bundle type for some reason is part of module Sales
We did this in the orders we have a composite, but still BundleCreditMemoItem is part of  Magento\SalesGraphQl